### PR TITLE
Add option to tokenize numbers per digit

### PR DESCRIPTION
**Pull request by johnflux**

### CODE REVIEW

**Comment by sanagno** [comment]
That is very cool! have you tested this at all?

File: model/model_training/utils.py [hunk 1]
```diff
@@ -183,6 +184,9 @@ def get_tokenizer(conf) -> transformers.AutoTokenizer:
     tokenizer = transformers.AutoTokenizer.from_pretrained(conf.model_name, cache_dir=conf.cache_dir)
     tokenizer_config = match_tokenizer_name(conf.model_name)
 
+    if tokenizer_config.per_digit_tokens:
```
Comment (sanagno) [inline model/model_training/utils.py:187]: also conf.per_digit_tokens

File: model/model_training/trainer_sft.py [hunk 2]
```diff
@@ -189,6 +190,7 @@ def argument_parsing(notebook=False, notebook_args=None):
     conf["wandb_entity"] = args.wandb_entity
     conf["local_rank"] = args.local_rank
     conf["deepspeed"] = args.deepspeed
+    conf["per_digit_tokens"] = args.per_digit_tokens
```
Comment (sanagno) [inline model/model_training/trainer_sft.py:193]: i would move this to the config.yaml instead of args

Comment (johnflux) [reply model/model_training/trainer_sft.py:193]: How about both?  Allowing command line to override it.

Comment (sanagno) [reply model/model_training/trainer_sft.py]: this is done by default

**Review by sanagno — APPROVED** [review]
Looks good

**Review by theblackcat102 — APPROVED** [review]
LGTM

File: model/reward/instructor/utils.py [hunk 3]
```diff
@@ -90,16 +94,21 @@ def argument_parsing(parser):
         "per_device_train_batch_size": 8,
         "gradient_accumulation_steps": 8,
         "gradient_checkpointing": False,
-        "deepspeed": args.deepspeed,
-        "local_rank": args.local_rank,
+        "deepspeed": False,
```
Comment (andreaskoepf) [inline model/reward/instructor/utils.py:97]: looks like a local testing change that should not be committed ...

**Comment by SummerSigh** [comment]
Seems good!

**Review by SummerSigh — APPROVED** [review]

**Pull request merged**